tests/test_client.py

```diff
@@ -1766,3 +1769,40 @@ def retry_handler(_request: httpx.Request) -> httpx.Response:
         ) as response:
             assert response.retries_taken == failures_before_success
             assert int(response.http_request.headers.get("x-stainless-retry-count")) == failures_before_success
+
+    def test_get_platform(self) -> None:
+        # Issue https://github.com/openai/openai-python/issues/1827 was caused
+        # asyncify leaving threads unterminated when used with nest_asyncio.
+        # Since nest_asyncio.apply() is global and cannot be un-applied, this
+        # test is run in a separate process to avoid affecting other tests.
+        test_code = dedent("""\
+        import asyncio
+        import nest_asyncio
+
+        import threading
+
+        from openai._base_client import get_platform
+        from openai._utils import asyncify
+
+        async def test_main() -> None:
+            result = await asyncify(get_platform)()
+            print(result)
+            for thread in threading.enumerate():
+                print(thread.name)
+
+        nest_asyncio.apply()
+        asyncio.run(test_main())
+        """)
+        with subprocess.Popen(
+            [sys.executable, "-c", test_code],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        ) as process:
+            try:
+                process.wait(2)
+                assert process.returncode == 0
+            except subprocess.TimeoutExpired:
+                process.kill()
+                assert 1 == 0, "calling get_platform using asyncify resulted in a hung process"
```

Comment:
```suggestion
                assert False, "calling get_platform using asyncify resulted in a hung process"
```
or
```suggestion
                raise AssertionError("calling get_platform using asyncify resulted in a hung process")
```
I think should avoid the lint error